containers/daap-python-base/src/var/task/data_product_metadata.py

```diff
@@ -495,3 +473,83 @@ def _detect_update_type(self) -> UpdateType:
     def add_auto_generated_metadata(self):
         """adds key value pairs to metadata that are not given by a user."""
         raise NotImplementedError
+
+
+class VersionCreator:
```

Comment:
Move this to a separate module. We can build on it in https://github.com/ministryofjustice/data-platform/issues/1925